Monytor.WebApi/Controllers/CollectorConfigController.cs

```diff
@@ -0,0 +1,85 @@
+            return Ok(await _collectorConfigService.GetCollectorConfigAsync(Uri.UnescapeDataString(collectorConfigId)));
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<string>> CreateCollectorConfig([FromBody]CreateCollectorConfigCommand command) {
+            var id = await _collectorConfigService.CreateCollectorConfigAsync(command);
+            return Ok(id);
+        }
+
+        [HttpDelete("{*collectorConfigId}")]
+        public async Task<ActionResult> DeleteCollectorConfigAsync(string collectorConfigId) {
+            await _collectorConfigService.DeleteCollectorConfigAsync(Uri.UnescapeDataString(collectorConfigId));
+            return Ok();
+        }
+
+        [HttpPost("{collectorConfigId}/AddSqlCountCollector")]
+        public async Task<ActionResult<string>> AddSqlCountCollector(string collectorConfigId, [FromBody] AddSqlCountCollectorToConfigCommand command) {
+            await _collectorConfigService.AddCollectorAsync(Uri.UnescapeDataString(collectorConfigId), command);
+            return Ok();
+        }
+
+        [HttpPost("{collectorConfigId}/AddRavenDbStartingWithCollector")]
+        public async Task<ActionResult<string>> AddRavenDbStartingWithCollector(string collectorConfigId, [FromBody] AddRavenDbStartingWithCollectorToConfigCommand command) {
+            await _collectorConfigService.AddCollectorAsync(Uri.UnescapeDataString(collectorConfigId), command);
+            return Ok();
+        }
+
+        [HttpPost("{collectorConfigId}/AddRavenDbCollectionCollector")]
+        public async Task<ActionResult<string>> AddRavenDbCollectionCollector(string collectorConfigId, [FromBody] AddRavenDbCollectionCollectorToConfigCommand command) {
+            await _collectorConfigService.AddCollectorAsync(Uri.UnescapeDataString(collectorConfigId), command);
+            return Ok();
+        }
+
+        [HttpPost("{collectorConfigId}/AddRavenDbAllCollectionCollector")]
+        public async Task<ActionResult<string>> AddRavenDbAllCollectionCollector(string collectorConfigId, [FromBody] AddRavenDbAllCollectionCollectorToConfigCommand command) {
+            await _collectorConfigService.AddCollectorAsync(Uri.UnescapeDataString(collectorConfigId), command);
+            return Ok();
+        }
+
+        [HttpPost("{collectorConfigId}/AddRestApiCollector")]
+        public async Task<ActionResult<string>> AddRavenDbAllCollectionCollector(string collectorConfigId, [FromBody] AddRestApiCollectorToConfigCommand command) {
+            await _collectorConfigService.AddCollectorAsync(Uri.UnescapeDataString(collectorConfigId), command);
+            return Ok();
+        }
+
+        [HttpPost("{collectorConfigId}/AddSystemInformationCollector")]
+        public async Task<ActionResult<string>> AddRavenDbAllCollectionCollector(string collectorConfigId, [FromBody] AddSystemInformationCollectorToConfigCommand command) {
```

Comment:
A little bit weird that we have for every collector a method to register...

Reply:
Yeah, but I did not like to have a more generic approach with empty fields. We will see how it feels when implementing a client against this.